src/cbmc_starter_kit/setup_ci.py

```diff
@@ -0,0 +1,163 @@
+    'rel-1.5.3' respectively).
+
+    To find semantic versions for these tools or tags for the solvers, visit:
+    - https://github.com/diffblue/cbmc/releases
+    - https://github.com/model-checking/cbmc-viewer/releases
+    - https://github.com/awslabs/aws-build-accumulator/releases
+    - https://github.com/arminbiere/kissat/tags
+    - https://github.com/arminbiere/cadical/tags
+    """
+    options = [{
+        'flag': '--github-actions-runner',
+        'required': True,
+        'metavar': "<ubuntu-20.04>|<name-of-your-Ubuntu-20.04-large-runner>",
+        'help': """
+                The name of the Ubuntu 20.04 GitHub-hosted runner that will run
+                proofs inside of GitHub Actions. For example: 'ubuntu-20.04'
+                If you have created a large runner for your repo, you can
+                specify its name."""
+        }, {
+        'flag': '--cbmc',
+        'metavar': '<latest>|<X.Y.Z>',
+        'default': 'latest',
+        'help': "Use this version of CBMC in CI. default: %(default)s"
+        }, {
+        'flag': '--cbmc-viewer',
+        'metavar': '<latest>|<X.Y>',
+        'default': 'latest',
+        'help': "Use this version of CBMC viewer in CI. default: %(default)s"
+        }, {
+        'flag': '--litani',
+        'metavar': '<latest>|<X.Y.Z>',
+        'default': 'latest',
+        'help': "Use this version of Litani in CI. default: %(default)s"
+        }, {
+        'flag': '--kissat',
+        "metavar": "<latest>|<TAG>",
+        'default': 'latest',
+        'help': "Use this tag of kissat in CI. default: %(default)s"
+        }, {
+        'flag': '--cadical',
+        "metavar": "<latest>|<TAG>",
+        'default': 'latest',
+        'help': "Use this tag of cadical in CI. default: %(default)s"
+        }]
+    args = arguments.create_parser(
+        options=options,
+        description=desc,
+        formatter_class=RawDescriptionHelpFormatter).parse_args()
+    arguments.configure_logging(args)
+    return args
+
+################################################################
+
+def _read_file_template(path):
+    """Read file and return list of lines"""
+    with open(path, encoding='utf-8') as data:
+        return data.read().splitlines()
+
+def _write_file_template(lines, path):
+    """Write file at specified path with the provided list of lines"""
+    with open(path, "w", encoding='utf-8') as data:
+        data.write('\n'.join(lines) + '\n')
+
+def _replace_placeholders_in_workflow_template(lines, replacements):
+    """Returns a list of new lines, where some lines have had a placeholder
+    value appropriately updated."""
+    buf = []
+    cmpl = re.compile(r"\s+(?P<key>[\w_]+):\s<__[\w_]+__>")
+    for line in lines:
+        if not line or line[len(line) - 1] != ">":
+            # Append immediately if empty line or line does not end with '>'
+            buf.append(line)
+            continue
+        if line.strip() == "runs-on: <__GITHUB_ACTIONS_RUNNER__>":
```

Comment:
Our compiled regex pattern is:

```python
cmpl = re.compile(r"\s+(?P<key>\w+):\s<__\w+__>")
```

Note the `key`. If you look at `patch_proof_ci_workflow(args, proof_ci_path)` you will see a `replacements` dict. The key in each entry of this dictionary is actually the same  as the `key` in the matched group above.

The regex won't work because `runs-on` is not a key.